src/main/java/com/gw/tasks/TaskManager.java

```diff
@@ -168,21 +168,24 @@ public synchronized void notifyWaitinglist() {
     logger.debug("notify waiting list to pay attention to the released worker");
     if (waitinglist.size() > 0 && wm.getCurrentWorkerNumber() < worknumber) {
       orderWaitingList();
+
+      // Process tasks iteratively
       for (int i = 0; i < waitinglist.size(); i++) {
         GeoweaverProcessTask newtask = (GeoweaverProcessTask) waitinglist.get(i);
         if (newtask.getIsReady()) {
-          waitinglist.remove(newtask);
-          if (newtask.checkShouldPassOrNot()){
-            newtask.endPrematurely();
-            notifyWaitinglist();
-          }else{
-            executeATask(newtask);
+          waitinglist.remove(newtask); // Remove task from waiting list
+          if (newtask.checkShouldPassOrNot()) {
+            newtask.endPrematurely(); // End the task immediately if it should not proceed
```

Comment:
you removed the notifyWaitinglist function, please explain why? how does the queue know that this waiting list is changed for worker to pick up new one?